test/threads.jl

```diff
@@ -141,3 +141,40 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    data1::Atomic{Int}
+    data2::Atomic{Int}
+    correct::Bool
+    CommBuf() = new()
+end
+function test_fence_write(commbuf::CommBuf)
+    for i in 10000:-1:1
+        commbuf.data1[] = i
+        fence()
+        commbuf.data2[] = i
```

Comment:
This fence is a no-op on x86 -- so no, I couldn't observe failure yet. We'll have to test on a different architecture.